wallet/src/Pos/Wallet/Web/Methods/Misc.hs

```diff
@@ -101,3 +111,22 @@ testResetAll = deleteAllKeys >> testReset
     deleteAllKeys = do
         keyNum <- length <$> getSecretKeys
         replicateM_ keyNum $ deleteSecretKey 0
+
+----------------------------------------------------------------------------
+-- Print wallet state
+----------------------------------------------------------------------------
+
+data WalletStateSnapshot = WalletStateSnapshot
+    { wssWalletStorage :: WalletStorage
+    } deriving (Generic)
+
+deriveJSON defaultOptions ''WalletStateSnapshot
+
+instance MimeRender OctetStream WalletStateSnapshot where
+    mimeRender _ = encode
+
+instance Buildable WalletStateSnapshot where
+    build _ = "<wallet-state-snapshot>"
+
+dumpState :: MonadWalletWebMode m => m WalletStateSnapshot
+dumpState = WalletStateSnapshot <$> getWalletStorage
```

Comment:
Is it ok to dump the whole storage? Doesn't it contain sensitive data?

Reply:
It contains addresses and they are sensitive, but we got such sensitive data in logs when we were investigating accidents and we decided it's not a big deal